### DISCUSSION TITLE
Sorting in One to One Relation is not working

### DISCUSSION THREAD

**Discussion by Sarthak-Bhadiyadra** [supabase/supabase#19273] [Questions] [answered]
Dear Supabase Support Team,

I am currently experiencing an issue related to sorting within a one-to-one relation while using Supabase. In the context of my application, I am attempting to implement sorting functionality based on various fields associated with related tables, such as **users**, **dishes**, **sub_categories**, and **restaurants**.

The problem arises when applying sorting conditions to fields associated with these related tables; the sorting mechanism does not seem to be functioning as expected. I have reviewed the code and attempted to sort by different fields, but the results are not consistent or accurate.

**Key points regarding the issue:**

Sorting Logic: I've implemented sorting based on fields from associated tables using the order method with foreign tables. However, despite following the documented guidelines, the sorting results are inconsistent.

Debugging Efforts: I've extensively tested sorting for relational individual fields to isolate the problem. However, the expected sorting order is not reflected accurately but when i use the root table's (for here my main table is "report" ) columns then it's working (i.e. description , status,etc... ) (for more info see the example below)

Verification: I've verified the column names, relationships between tables, and ensured that the sorting is applied before pagination to obtain the desired sorted subset of data.

Potential Null Values: I've checked for null values in fields used for sorting and handled them appropriately. However, the presence of null values does not seem to be the cause of the issue.

**Client Side(JS library) Query :**

``` 
export const fetchAllReports = async (
  restaurant_id: string | null,
  from: number,
  to: number,
  searchKeyword: string,
  sorting: { id: string; desc: boolean }
) => {
  let query = supabase
    .from(report_table)
    .select(
      "*,users(first_name,full_name,avtar,id),sub_categories(name),dishes(name,dish_media(medias(media_url))),restaurants(name,logo),status",
      { count: "exact" }
    )
    .eq("dishes.dish_media.is_official", true)
    .eq("dishes.dish_media.status", "accept")
    .limit(1, {
      foreignTable: "dishes.dish_media",
    })
    .range(from, to);
  if (restaurant_id) {
    query = query.eq("restaurant_id", restaurant_id);
  }
  if (searchKeyword !== "") {
    let searchString = `%${searchKeyword.trim()}%`;

    query = query.or(`description.ilike.${searchString}`);
  }

  if (!sorting) {
    query = query.order("created_at", { ascending: false });
  }

  if (sorting) {
    if (sorting.id === "users_name") {
      query = query.order("first_name", {
        foreignTable: "users",
        ascending: !sorting.desc,
      });
    } else if (sorting.id === "dishes_name") {
      query = query.order("name", {
        foreignTable: "dishes",
        ascending: !sorting.desc,
      });
    } else if (sorting.id === "description") {
      query = query.order("description", { ascending: !sorting.desc });
    } else if (sorting.id === "type") {
      query = query.order("type", { ascending: !sorting.desc });
    } else if (sorting.id === "categories") {
      query = query.order("name", {
        foreignTable: "sub_categories",
        ascending: !sorting.desc,
      });
    } else if (sorting.id === "restaurants_name") {
      query = query.order("name", {
        foreignTable: "restaurants",
        ascending: !sorting.desc,
      });
    } else if (sorting.id === "categories") {
      query = query.order("name", {
        foreignTable: "sub_categories",
        ascending: !sorting.desc,
      });
    } else if (sorting.id === "status") {
      query = query.order("status", {
        ascending: sorting.desc,
      });
    }
  }
  return await query;
}
 ```

I would greatly appreciate any guidance, insights, or assistance from your team to address this matter. It's crucial for our application's functionality to have consistent and accurate sorting within these one-to-one relations.

Please let me know if there are any additional details or specific tests I can perform to further diagnose and resolve this issue. Your prompt attention and support in resolving this matter would be immensely valuable.

Thank you for your time and assistance.

Warm regards,

Bhadiyadra Sarthak

**Comment by GaryAustin1** [comment, marked as answer]
I’m not with the Supabase team but I think you are trying to sort the parent table by the related tables value.  If so the method you are using only sorts the embedded tables themselves.  
https://supabase.com/blog/postgrest-11-prerelease This link has a newer method to sort the parent table by the embedded tables.  Basically you do .order(‘tablename(colname)’).

**Reply by Sarthak-Bhadiyadra** [reply]
Yes , It's Working with the newer method . Thank You  @GaryAustin1
